src/main/java/org/javamodularity/moduleplugin/tasks/MergeClassesTask.java

```diff
@@ -0,0 +1,39 @@
+package org.javamodularity.moduleplugin.tasks;
+
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.ApplicationPlugin;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.specs.Spec;
+import org.javamodularity.moduleplugin.extensions.CompileModuleOptions;
+
+import java.util.List;
+
+public class MergeClassesTask extends AbstractModulePluginTask {
+    public MergeClassesTask(Project project) {
+        super(project);
+    }
+
+    public void configureMergeClasses() {
+        var mergeClasses = helper().mergeClassesTask();
+
+        mergeClasses.into(helper().getMergedDir());
+
+        mergeClasses.onlyIf(new Spec<Task>() {
+            @Override
+            public boolean isSatisfiedBy(Task element) {
+                return helper().isMergeRequired();
+            }
+        });
+
+        List.of(CompileTask.COMPILE_KOTLIN_TASK_NAME,
+                JavaPlugin.COMPILE_JAVA_TASK_NAME,
+                CompileTask.COMPILE_GROOVY_TASK_NAME,
+                CompileModuleOptions.COMPILE_MODULE_INFO_TASK_NAME
+        ).forEach(name -> helper().findTask(name).ifPresent(mergeClasses::dependsOn));
```

Comment:
I just wanted to say how much I like this piece of code! It reads like prose: "for each of these tasks, if it's present, make `mergeClasses` depend on it" 🙂

Reply:
🙂